xarray/core/common.py

```diff
@@ -744,6 +745,35 @@ def close(self):
             self._file_obj.close()
         self._file_obj = None
 
+    def isin(self, test_elements):
+        """Tests each value in the array for whether it is in the supplied list
+        Requires NumPy >= 1.13 
+
+        Parameters
+        ----------
+        element : array_like
+            Input array.
+        test_elements : array_like
+            The values against which to test each value of `element`.
+            This argument is flattened if it is an array or array_like.
+            See notes for behavior with non-array-like parameters.
+
+        -------
+        isin : same as object, bool
+            Has the same shape as object
+        """
+        if LooseVersion(np.__version__) < LooseVersion('1.13.0'):
+            raise ImportError('isin requires numpy version 1.13.0 or later')
+        from .computation import apply_ufunc
+
+        return apply_ufunc(
+            np.isin,
+            self,
+            kwargs=dict(test_elements=test_elements),
```

Comment:
It's probably a better idea to explicitly unwrap `.data` from `test_elements` if it's an xarray object, and explicitly raise for `xarray.Dataset`. Otherwise numpy will probably give a really strange error message.

Reply:
Indeed, see #2032 for what converting a Dataset to a numpy array does.

Reply:
I merged your branch - is that better than an additional check? 

Why extract `.data` - won't the standard machinery take care of that? I added a test

Reply:
I suppose you're right, the standard machinery will work fine here for now. In the future when dask supports isin (https://github.com/dask/dask/pull/3363) we'll want to use `.data` so we can keep it as a dask array.